### ISSUE TITLE
grapher: investigate usage of flot

### ISSUE THREAD

**Issue by tiborsimko** [inveniosoftware/invenio#20]
Originally on 2010-04-26

1) Instead of using gnuplot in the grapher, investigate the usage of
jQuery plotting tools, such as [http://code.google.com/p/flot/].
Advantage being the possibility to zoom into graphs easily.
Disadvantage may be higher server side stress related to Ajax.  So we
would have to nicely cache the {(X1,Y1),(X2,Y2),...} JSON structures
that Flot would need, just as we are now caching the PNGs produced by
gnuplot.  The server stress would have to be tested.  Caching PNGs is
definitely more scalable: once cached, they are served directly by
Apache, not by Invenio WSGI application.

2) Flot would be especially well suited for admin-like graphs in the
WebStat module, where currently the x-axis scale is chosen from a
drow-down selection box.  It would be even more practical to zoom
here, and the stress issue would be less important here, since there
is only a few connections to the admin-level pages, as opposed to the
user-level pages.
